packages/java-edition/src/json/checker/index.ts

```diff
@@ -1,18 +1,50 @@
 import type * as core from '@spyglassmc/core'
 import * as json from '@spyglassmc/json'
+import type * as mcdoc from '@spyglassmc/mcdoc'
 import { dissectUri } from '../../binder/index.js'
 
+function createTagDefinition(registry: string): mcdoc.McdocType {
+	const id: mcdoc.AttributeValue = {
+		kind: 'tree',
+		values: {
+			registry: {
+				kind: 'literal',
+				value: { kind: 'string', value: registry },
+			},
+			tags: {
+				kind: 'literal',
+				value: { kind: 'string', value: 'allowed' },
+			},
+		},
+	}
+	return {
+		kind: 'concrete',
+		child: { kind: 'reference', path: '::java::data::tag::Tag' },
+		typeArgs: [{ kind: 'string', attributes: [{ name: 'id', value: id }] }],
+	}
+}
+
 export const file: core.Checker<json.JsonFileNode> = (node, ctx) => {
 	const child = node.children[0]
 	if (ctx.doc.uri.endsWith('/pack.mcmeta')) {
-		return json.checker.definition('::java::pack::Pack')(child, ctx)
+		const type: mcdoc.McdocType = {
+			kind: 'reference',
+			path: '::java::pack::Pack',
+		}
+		return json.checker.index(type)(child, ctx)
```

Comment:
This seems a bit more messy, any reason to remove the `definition` shorthand?

Reply:
No particular reason 🤷 I don't think it's that messy to supply a mcdoc type.